main/ogn-aprs.go

```diff
@@ -0,0 +1,240 @@
+/*
+	Copyright (c) 2022 Quentin Bossard
+	Distributable under the terms of The "BSD New" License
+	that can be found in the LICENSE file, herein included
+	as part of this header.
+
+	aprs.go: Routines for reading traffic from aprs
+*/
+
+package main
+
+import (
+	"bufio"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var aprsOutgoingMsgChan chan string = make(chan string, 100)
+var aprsIncomingMsgChan chan string = make(chan string, 100)
+var aprsExitChan chan bool = make(chan bool, 1)
+
+
+func authenticate(c net.Conn) {
+	filter := ""
+	if mySituation.GPSFixQuality > 0 {
+		filter = fmt.Sprintf(
+			"filter r/%.7f/%.7f/%d\r\n", mySituation.GPSLatitude, mySituation.GPSLongitude, 100)
+	}
+	auth := fmt.Sprintf("user OGNNOCALL pass -1 vers stratux 0.28 %s\r\n", filter)
+	fmt.Printf(auth)
+	fmt.Fprintf(c, auth)
+}
+
+func keepalive(c net.Conn) {
+	ticker := time.NewTicker(240 * time.Second)
+
+	go func() {
+		for t := range ticker.C {
+			fmt.Fprintf(c, "# stratux keepalive %s\n", t)
+			fmt.Printf("# stratux keepalive %s\n", t)
+		}
+	}()
+}
+
+func updateFilter(c net.Conn) {
+	ticker := time.NewTicker(10 * time.Second)
+	go func() {
+		for range ticker.C {
+			if mySituation.GPSFixQuality > 0 {
+				filter := fmt.Sprintf(
+					"#filter r/%.7f/%.7f/%d\r\n", mySituation.GPSLatitude, mySituation.GPSLongitude, 100)
```

Comment:
I feel like 100km is far too much. It can be quite a lot of data on a sunny weekend, and I don't see any usefulness in that range. For traffic avoidance, 10km would be enough. If someone uses Stratux on their tower of an uncontroled airfield, maybe up to 30km, but more than that is just "showing off with only drawbacks in reality". Or am I missing something?

If you really see a purpose in that much range, maybe make it configurable in globalSettings (no need for web interface, just to have some way to at least configure it manually)

Reply:
449db4be125134457b43ce0abe9496591c4c46eb